hilti/toolchain/src/compiler/optimizer.cc

```diff
@@ -1097,21 +1101,24 @@ struct ConstantPropagationVisitor : OptimizerVisitor {
         visitor::visit(tv, n.value());
     }
 
-    void populate_dataflow(AnalysisResult& result, const ConstantMap& init) {
-        std::vector<detail::cfg::GraphNode> worklist;
+    void populate_dataflow(AnalysisResult& result, const ConstantMap& init, const ID& function_name) {
+        auto worklist = result.cfg.postorder();
+        assert(worklist.size() >= 1);
```

Comment:
Wondering whether a comment here would help (e.g., "we always expect at least begin/end nodes), but maybe just exiting before the first `pop_front` would be sufficiently self-explanatory.